gobblin-config-management/gobblin-config-core/src/main/java/gobblin/config/store/hdfs/SimpleHDFSConfigStoreFactory.java

```diff
@@ -0,0 +1,160 @@
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+
+import com.google.common.base.Strings;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileStatus;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.security.UserGroupInformation;
+
+import gobblin.config.store.api.ConfigStoreCreationException;
+import gobblin.config.store.api.ConfigStoreFactory;
+
+
+/**
+ * An implementation of {@link ConfigStoreFactory} for creating {@link SimpleHDFSConfigStore}s. This class only works
+ * the physical scheme {@link #HDFS_SCHEME_NAME}.
+ *
+ * @see {@link SimpleHDFSConfigStore}
+ */
+public class SimpleHDFSConfigStoreFactory implements ConfigStoreFactory<SimpleHDFSConfigStore> {
+
+  protected static final String SIMPLE_HDFS_SCHEME_PREFIX = "simple-";
+  protected static final String HDFS_SCHEME_NAME = "hdfs";
+
+  @Override
+  public String getScheme() {
+    return SIMPLE_HDFS_SCHEME_PREFIX + getPhysicalScheme();
+  }
+
+  /**
+   * Creates a {@link SimpleHDFSConfigStore} for the given {@link URI}. The {@link URI} specified should be the fully
+   * qualified path to the dataset in question. For example,
+   * {@code simple-hdfs://[authority]:[port][path-to-config-store][path-to-dataset]}. It is important to note that the
+   * path to the config store on HDFS must also be specified. The combination
+   * {@code [path-to-config-store][path-to-dataset]} need not specify an actual {@link Path} on HDFS.
+   *
+   * <p>
+   *   If the {@link URI} does not contain an authority, a default authority and root directory are provided. The
+   *   default authority is taken from the NameNode {@link URI} the current process is co-located with. The default path
+   *   is "/user/[current-user]/".
+   * </p>
+   *
+   * @param  configKey       The URI of the config key that needs to be accessed.
+   *
+   * @return a {@link SimpleHDFSConfigStore} configured with the the given {@link URI}.
+   *
+   * @throws ConfigStoreCreationException if the {@link SimpleHDFSConfigStore} could not be created.
+   */
+  @Override
+  public SimpleHDFSConfigStore createConfigStore(URI configKey) throws ConfigStoreCreationException {
+    FileSystem fs = createFileSystem(configKey);
+    URI physicalStoreRoot = getStoreRoot(fs, configKey);
+    URI logicalStoreRoot = URI.create(SIMPLE_HDFS_SCHEME_PREFIX + physicalStoreRoot);
+    return new SimpleHDFSConfigStore(fs, physicalStoreRoot, logicalStoreRoot);
+  }
+
+  /**
+   * Returns the physical scheme this {@link ConfigStoreFactory} is responsible for. To support new HDFS
+   * {@link FileSystem} implementations, subclasses should override this method.
+   */
+  protected String getPhysicalScheme() {
+    return HDFS_SCHEME_NAME;
+  }
+
+  /**
+   * Gets a default root directory if one is not specified. The default root dir is {@code /user/[current-user]/}.
+   */
+  protected Path getDefaultRootDir() throws IOException {
```

Comment:
We talked to Chavdar, and he suggested that the `ConfigStoreFactory` needs some type of `configure(Config config)` method so that it can configure itself. This `Config` object would also contain reasonable defaults. However, since that method currently does not exist, we decided just to stick with this for now.